### ISSUE TITLE
Make Unit Tests run in a single thread

### ISSUE THREAD

**Issue by maceligueta** [KratosMultiphysics/Kratos#1807]
I just measured the time it takes to run 4 simple Unit Tests of the DEM app. More than 8 minutes in parallel (8 threads) and 2 seconds in a single thread!
I am afraid that putting them in the nightly builds would be a bad idea, until we clarify how to make it run in a single processor.
How should we proceed? 
os.environ['OMP_NUM_THREADS']='1' looks like an ugly option, but it would work, I think.
Any idea is welcome.

**Comment by josep-m-carbonell** [comment]
I don't know if they are in python, maybe something like this is possible:
parallel = KratosMultiphysics.OpenMPUtils()
num_treads = GetNumThreads()
parallel.SetNumThreads(1)
self.test.Run()
parallel.SetNumThreads(num_threads)

**Comment by philbucher** [comment]
@maceligueta I also experienced these problems, but I suggest using two threads bcs some error checkings are only active if run with more than one thread (see e.g. data_value_container.h)

**Comment by pooyan-dadvand** [comment]
I would dig to see what this is happening....

**Comment by philbucher** [comment]
I think since the tests are small, the overhead caused by omp is larger than the benefit

**Comment by RiccardoRossi** [comment]
i think that overhead alone does not justify passing from seconds to minutes.

I guess there is some quadratic or cubic behaviour coming into play...

**Comment by maceligueta** [comment]
@RiccardoRossi  they are tests with 1 to 3 spheres, not much to compute, and I don't see how quadratic or cubic behavior can affect. At the DEM app we have many loops over the model part in each time step, and many, many time steps. I am pretty sure that the OMP overhead is the cause.

**Comment by RiccardoRossi** [comment]
what can i say... it is very strange to me an overhead of minutes. I haven't seen this happening before.

**Comment by philbucher** [comment]
I also had this problem

**Comment by maceligueta** [comment]
@RiccardoRossi  I just compiled Kratos in RWDInfo mode and loaded the profiler to find the source of time loss but I could not track it: now the multi-thread version takes 10 seconds (still 2 seconds for the single thread) to run. Then I went back to release and the same results 10s vs 2s. I can't explain this behavior, because yesterday I left mi PC compiling the release mode and ran the tests when I got home, so I am pretty sure that those times were measured well. Anyways, now it is working fine, so I close the issue and will re-open if it happens again.
